src/tsolvers/lrasolver/LRASolver.C

```diff
@@ -191,6 +191,32 @@ LRASolver::~LRASolver( )
 
 LRALogic&  LRASolver::getLogic() { return logic; }
 
+lbool LRASolver::getPolaritySuggestion(PTRef ptref) const {
+    if (!this->isInformed(ptref)) { return l_Undef; }
+    LVRef var = this->lavarStore.getVarByLeqId(logic.getPterm(ptref).getId());
+    if (!model.hasModel(var)) { return l_Undef; }
+    LABoundRefPair bounds = this->boundStore.getBoundRefPair(ptref);
+    auto const& val = model.read(var);
+    bool positive = false;
+    auto const& positive_bound = this->boundStore[bounds.pos];
+    if ((positive_bound.getType() == bound_l && positive_bound.getValue() <= val)
+        || (positive_bound.getType() == bound_u && positive_bound.getValue() >= val)) {
+        positive = true;
+    }
+    bool negative = false;
+    auto const& negative_bound = this->boundStore[bounds.neg];
+    if ((negative_bound.getType() == bound_l && negative_bound.getValue() <= val)
+        || (negative_bound.getType() == bound_u && negative_bound.getValue() >= val)) {
+        negative = true;
+    }
+    // MB: Maybe we will not get any suggestion, if the value is <0|-1/2>, but the bounds are <0|0> and <0|-1>
+//    assert(positive || negative);
```

Comment:
What's the invariant here?  Can assert(positive || negative) be removed?

Reply:
Originally I thought that the invariant would be positive XOR negative (exactly one will be true and the other one false).
But then I ran into a situation where a value of a variable, "x", was <0, -1/2> (-1/2 is the delta), and the constraint was x >= 0. The positive bound on this is a lower bound with value <0,0> and the negative bound is an upper bound with value <0, -1> ( x < 0, i.e, x <= 0 - Delta)
In this case the current value does not satisfy neither positive nor negative bound.
It is kind of a special case but I had to comment out the assert because of this since it is a valid situation.
I just wanted to stress that this assertion is not valid (even though I originally thought it would be) so I left it there as a comment.

Do you suggest to remove it completely?

Reply:
Hm ok... In that case the 'consistent' suggestion would be to set negative polarity, that is, x < <0,-1>?  But it wouldn't really be consistent in the end because x = <0,-1/2> would break x < <0,-1>.   What does our code do in that case?

I'm not sure what's the right thing to do about the assert.  Perhaps removing it and expanding the comment a little as you do above would be better.

Reply:
Yes, neither choice is consistent with the assignment, so the theory solver just says "I don't know" and the decision falls back to the SAT solver.

I will remove the commented assert and add more detailed explanation.